src/meta/src/rpc/server.rs

```diff
@@ -202,6 +212,11 @@ pub async fn rpc_serve_with_store<S: MetaStore>(
             }
         }
 
+        // TODO: Enabling this causes the recovery test to fail
+        // See https://buildkite.com/risingwavelabs/pull-request/builds/14686#01855869-b9d7-432b-9ea8-b835830f1a8e
+        // tracing::info!("Waiting, to give former leaders fencing mechanism time to trigger");
+        // sleep(Duration::from_millis(lease_interval_secs * 1000 + 500));
+
```

Comment:
I would have liked to introduce a break here to give the fencing mechanism time to kill the old leader before the new leader takes over. Unfortunately this seems to break the recovery test. Should we still try to introduce a wait time here @yezizp2012 ?

Reply:
Small correction from above statement: We do not elect a new leader directly after the lease expires. We give the old leader a little extra time. The current implementation looks like this: 

```rust 
// delete lease and run new election if lease is expired for some time
let some_time = lease_time_sec / 2;
let (_, lease_info) = leader_lease_result.unwrap();
if lease_info.get_lease_expire_time() + some_time < since_epoch().as_secs() {
    tracing::warn!("Detected that leader is down");
```